cpp/src/arrow/array/builder_binary.h

```diff
@@ -372,6 +372,8 @@ class BaseBinaryBuilder : public ArrayBuilder {
     return util::string_view(reinterpret_cast<const char*>(value_data), value_length);
   }
 
+  util::string_view operator[](int64_t i) const { return GetView(i); }
```

Comment:
@pitrou Builder's don't need Iterator and Iterator based overload I suppose or do they ?

Reply:
No, but I don't think they need `operator[]` either. This PR should only touch array classes, IMHO.